bionic/task_state.py

```diff
@@ -317,3 +329,54 @@ def _check_accessors_for_version_problems(self):
 
         for accessor in accessors_needing_saving:
             accessor.update_provenance()
+
+    # NOTE: This can be optimized further.
+    def new_state_for_completion(self, new_task_states_by_key, to_be_computed=True):
+        """
+        Returns a copy of task state after keeping only the necessary data
+        required for completion. Along with keeping only necessary data for
+        completion, this also removes all the memoized results since they can
+        be expensive to serialize.
+
+        Mainly used to reduce IPC overhead when sending the state over to a
```

Comment:
It's not just overhead, but also some things are simply impossible to serialize, right?

Reply:
oh yeah, good point. I'll call them both out here.